src/test/scala/com/microsoft/hyperspace/index/E2EHyperspaceRulesTests.scala

```diff
@@ -588,7 +693,7 @@ class E2EHyperspaceRulesTests extends QueryTest with HyperspaceSuite {
   private def verifyQueryPlanHasExpectedRootPaths(
       optimizedPlan: LogicalPlan,
       expectedPaths: Seq[Path]): Unit = {
-    assert(getAllRootPaths(optimizedPlan) === expectedPaths)
+    assert(getAllRootPaths(optimizedPlan).sortBy(_.getName) === expectedPaths.sortBy(_.getName))
```

Comment:
can we use toSet() here?

Reply:
There can be multiple same rootPaths from join query tests.